core/src/main/java/com/nextdoor/bender/handler/BaseHandler.java

```diff
@@ -373,6 +374,8 @@ public void run() {
       try {
         String raw = null;
         raw = this.ser.serialize(this.wrapper.getWrapped(ievent));
+        Stat bytesSent = new Stat("serializer.serialized_bytes", raw.getBytes(StandardCharsets.UTF_8).length);
```

Comment:
So there's a bit of a problem here that stems from the fact that the serialized payload is `String` when in fact it shouldn't be and something we really ought to fix. The problem here and other places in the code is `String.getBytes` will actually create a whole new copy of the data because java strings are immutable. Unfortunately `gson.toJson` returns a string and I leaked that when I probably shouldn't have. While it's not perfect I propose for now we use `raw.length()` which is close enough until we can do a refactor.

The other thing here is the creation of `Stat` needs to be done outside of the `map` action because the monitor doesn't perform summing on duplicate `Stat`. Checkout the `writeStats` function below. I think that's a more appropriate location.

Last thing, I know it's not documented very well but you'll want to use `addInvocationStat`. The `addInstanceStat` is meant for more of the entirety of the function where as `addInvocationStat` is for the current invocation.

Reply:
I'm fine with changing it, just wondering if your concern with .getBytes is unnecessarily creating another object? And sure I'll move the addition likely to writeStats.

Reply:
Yes, I prefer we not change the memory profile by adding a `getBytes`